dev/release-notebooks/en/3.4-release-notebook.md

```diff
@@ -0,0 +1,55 @@
+- The usage of the `Sokil\IsoCodes\IsoCodesFactory` class, which we've been using to retrieve lists of languages, currencies, countries and scripts, has been centralized into the facade `PKP\facades\Locale`, respectively by the methods `getLanguages()`, `getLanguages()`, `getCurrencies()`, `getCountries()`, and `getScripts()`.
+- The file `lib/pkp/registry/locales.xml` has been removed, now the list of locales is built upon the locale folders available for the application (e.g. `pkp-lib/locale/`) and the metadata information, such as display name, right-to-left script, etc., is retrieved programatically.
+- The `AppLocale` methods related to conversions between locale formats (ISO 639-1, ISO 639-2b, ...) have been moved into the class `PKP\i18n\LocaleConversion`.
+- The `AppLocale` methods used to retrieve locale metadata have been moved to the class `PKP\i18n\LocaleMetadata` (e.g. `getLocaleDirection()`, `isLocaleComplete()`).
+- The method `AppLocale::getAllLocales()` has been replaced by `PKP\facades\Locale::getLocales()`, but instead of returning the locale display name, it returns a `LocaleMetadata` instance, which holds extra information.
+- The following methods have been removed from the `AppLocale` class: `getLocaleStyleSheet()`, `getLocalePrecedence()`, `getLocaleFiles()`, `loadLocaleList()`, `addOctothorpes()`, `initialize()`, `makeComponentMap()`, `getFilenameComponentMap()`, `registerLocaleFile()`, `getLocaleStyleSheet()`, `isLocaleWithFamilyFirst()`, `reloadLocale()`, `getParameterNames()`, and `getTimeZone()`.
+- The translation hook has been renamed from `PKPLocale::translate` to `Locale::translate`, and its arguments have been modified as well.
+
+## General updates
+
+- The global function `__p(string $key, int $number, array $replace = [], ?string $locale = null)` has been introduced to translate pluralized locale keys.
+- Laravel's caching system has been setup with an OP Cache implementation.
+- Introduced the static method `APP\core\Application::isUpgrading()` to replace the constant `RUNNING_UPGRADE`.
+- Introduced the static method `APP\core\Application::isInstalled()` as a shortcut to `Config::getVar('general', 'installed')`.
+- Introduced the static method `APP\core\Application::isUnderMaintenance()` to check if the application is installed and not being upgraded.
+- Introduced the static method `PKP\session\SessionManager::isDisabled()` to replace the constant `SESSION_DISABLE_INIT`.
+
+## Smarty templates
+
+- The `translate` tag has been extended to support pluralized locale keys, the new syntax is as follows: `{translate key="localization.key.name" count="10" paramName="paramValue"}`, it's also possible to feed a `locale` param.
+- The `translate` modifier has also been extended to support pluralized locale keys, this is the new syntax: `{$foo|translate:count:123:locale:pt_BR:paramName:paramValue}`.
```

Comment:
Thanks for doing the release notebook. I don't want to add this until we're closer to 3.4's release, though, because things might change. Can you file an issue in the docs repo and copy this content into it? I'll then make sure it gets into the release notebook.

Reply:
As I didn't link it nowhere and also added a noindex, I thought it was ok to leave it as draft 😁